### PR TITLE
fix: console error when creating a list with inline style (close #221)

### LINKED ISSUE #221: console error when creating a list on a inline styled line
## Version

1.2.2

## Test Environment

ALL

## Current Behavior

1. open WYSIWYG editor
2. type below
image
3. set caret at the end of the text
4. click `ul` to make an unordered list
5. see the console error

It is caused by trying to set wrong start/end container of the selection.

## Expected Behavior

There should be no error, and set the selection correctly.

### PR DESCRIPTION
**Pull request by kyuwoo-choi**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has a description of the breaking change

### Description
#221

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by seonim-ryu** [comment]
[6/26] 저도 이견 없습니다. 고생하셨어요~

File: src/js/wysiwygCommands/ul.js [hunk 1]
```diff
@@ -40,9 +41,14 @@ const UL = CommandManager.command('wysiwyg', /** @lends UL */{
...
+    const newStartContainer = domUtil.containsNode(newLIs[0], startContainer)
+      ? startContainer : newLIs[0];
+    const newEndContainer = domUtil.containsNode(newLIs[newLIs.length - 1], endContainer)
+      ? endContainer : newLIs[newLIs.length - 1];
+
     range = sq.getSelection();
-    range.setStart(newLIs[0].firstChild, startOffset);
-    range.setEnd(newLIs[newLIs.length - 1].firstChild, endOffset);
+    range.setStart(newStartContainer, startOffset);
+    range.setEnd(newEndContainer, endOffset);
```
[the remark this answers is not in the document]
Comment (kyuwoo-choi) [reply src/js/wysiwygCommands/ul.js]: 옙!

File: src/js/domUtils.js [hunk 2]
```diff
@@ -294,6 +294,23 @@ const getPrevTextNode = function(node) {
...
 
+/**
+ * test whether root contains the given node
+ * @param {HTMLNode} root - root node
+ * @param {HTMLNode} node - node to test
+ * @returns {Boolean} true if root contains node
+ */
+const containsNode = function(root, node) {
+  const walker = document.createTreeWalker(root, 4, null, false);
+  let found = root === node;
```
Comment (shiren) [inline src/js/domUtils.js]: 괄호 하나 쳐주시면 읽기 좋을것 같아요

File: src/js/domUtils.js [hunk 3]
```diff
@@ -294,6 +294,23 @@ const getPrevTextNode = function(node) {
...
+
+  while (!found && walker.nextNode()) {
+    found = walker.currentNode === node;
```
Comment (shiren) [inline src/js/domUtils.js]: 여기도

File: src/js/wysiwygEditor.js [hunk 4]
```diff
@@ -640,6 +641,25 @@ class WysiwygEditor {
...
+  /**
+   * set selection by start/end container/offset
+   * @param {HTMLNode} startContainer - start container
+   * @param {Number} startOffset - start offset
+   * @param {HTMLNode} endContainer - end container
+   * @param {Number} endOffset - end offset
+   * @returns {Range} - range instance
+   * @memberof WysiwygEditor
+   */
+  setSelectionByContainerAndOffset(startContainer, startOffset, endContainer, endOffset) {
```
Comment (shiren) [inline src/js/wysiwygEditor.js]: 이 친구는 TC 만들기 쉬울것 같은데 빠져있네요?

**Pull request merged**